### ISSUE TITLE
etl::span of fixed size can be created from an array of a different size 

### ISSUE THREAD

**Issue by mnaydenov** [ETLCPP/etl#486]
```
#include <etl/span.h>
#include <span>

void f(etl::span<char, 25> )
{}

int main()
{
  std::array<char, 10> c;
  
  f(c);
  
  return 0;
}
```
The above code should **not** compile. 
The major attraction of using `span` instead of `array_view` is the fact, one can have it with fixed size, enforceable in compile time. This way if a function accepts a serial number of, say 25 digits, it can force the caller to give it only strings of this size. Currently however this enforcement does not work and the above code compiles. Using `std::span` works as expected and it fails to compile.

**Comment by jwellbelove** [comment]
Fixed 20.22.0

**Comment by jwellbelove** [comment]
Fixed 20.22.0
